hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
I agree with @horsburgh.  In addition to an e-mail to admins, we should make sure that we can see, via the admin console, information about, at least, the most recent DOI acquisition attempt. 

I don't know that we need to have all DOI attempts happen at the same time everyday.  Instead it might be simpler/more reliable to, on failure, re-submit the Crossref request task to the Celery queue a certain number of seconds into the future (say 7200, i.e. two hours).